### PR TITLE
Add a couple of linkcheck exceptions:

### PR DESCRIPTION
**Pull request by marxin**
Addresses:
```

Server returned 500 Internal Server Error for https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3AS-inactive+is%3Aclosed
Server returned 401 Unauthorized for https://objects.githubusercontent.com/github-production-release-asset-2e65be/343907537/60cd5880-7b81-11eb-9502-68d5f473ebe2?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241103%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241103T085906Z&X-Amz-Expires=300&X-Amz-Signature=e532fd7d2bd271f12988982bc8e5a47a0f966b40452b55d444bbb74292ac1fb1&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drustc.generic.wpaProfile&response-content-type=application%2Foctet-stream
Server returned 500 Internal Server Error for https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3AS-inactive
```

### CODE REVIEW

File: book.toml [hunk 1]
```diff
@@ -43,8 +43,14 @@ exclude = [
     "objects\\.githubusercontent\\.com",
     # The bug listing URL works only if an user is logged in, otherwise one gets 404.
     "github\\.com/issues\\?",
+    # Similarly 500 is sometimes returned here.
+    "github\\.com/rust-lang/rust/pulls\\?",
```
Comment (jieyouxu) [inline book.toml:47]: I'm not sure we want to skip this, isn't PR number linking kinda important

**Review by jieyouxu — COMMENTED** [review]
I'm not sure about the PR number exceptions, but I also am not sure about the linkcheck in general... so maybe cc @tshepang and @camelid?

File: book.toml (hunk 1 above)
Comment (marxin) [reply book.toml]: Yeah, it is. I want to skip only the 2 particular queries that bring the troubles:
```
https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3AS-inactive+is%3Aclosed
https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3AS-inactive
```

Does it make sense?

**Comment by marxin** [comment]
> I'm not sure about the PR link exceptions, but I also am not sure about the linkcheck in general... so maybe cc @tshepang and @camelid?

Note these are the last missing pieces after which the full link check will become green again. Moreover, with `mdbook-linkcheck2`, we can also enable checking for modified files (with `-f` option) and should result in more stable links in the future.

File: book.toml (hunk 1 above)
Comment (jieyouxu) [reply book.toml]: Oh okay, that makes sense. But also at the same time, it's horrifying(?) that these endpoints can return internal server errors?

Comment (jieyouxu) [reply book.toml]: @marxin out of curiosity, do you know if that happens frequently or did it occur during GitHub outages?

Comment (marxin) [reply book.toml]: If I create a dummy book and include the link, then it's fine. But these links are verified in a batch with other GH links (in parallel) and maybe we hit some kind of DoS protection mechanism. Who knows :)

Comment (marxin) [reply book.toml]: > @marxin out of curiosity, do you know if that happens frequently or did it occur during GitHub outages?

In my case, it happens every single time. Similarly, if you take a look at the nightly CI:
https://github.com/rust-lang/rustc-dev-guide/actions/runs/11653521916/job/32445996122

Comment (jieyouxu) [reply book.toml]: I suspect they're actually just rate limits, but yeah anyway, thanks for the info.

Comment (marxin) [reply book.toml]: The rate limits are properly reflected in the return code: `Server returned 429 Too Many Requests for https://github.com`, but these 500s are somehow different.

**Comment by marxin** [comment]
May I ping this PR @camelid as the `mdbook-linkcheck2` PR has been merged?

File: book.toml [hunk 2]
```diff
@@ -42,9 +42,15 @@ exclude = [
     "marketplace\\.visualstudio\\.com",
     "objects\\.githubusercontent\\.com",
     # The bug listing URL works only if an user is logged in, otherwise one gets 404.
-    "github\\.com/issues\\?",
+    "github\\.com/issues\\?q=.*",
+    # Similarly 500 is sometimes returned here.
+    "github\\.com/rust-lang/rust/pulls\\?q=.*",
```
Comment (jieyouxu) [inline book.toml:47]: Question: is this still a problem? Occassionally github does have outages, but it seems surprising that basic queries like that can return 500.

File: book.toml [hunk 3]
```diff
@@ -42,9 +42,15 @@ exclude = [
...
+    # 401 is returned here for unknown reason
+    "github\\.com/wesleywiser/rustc-bootstrap-wpa-analysis",
```
Comment (jieyouxu) [inline book.toml:49]: Question: similarly, is this still a problem?

**Comment by jieyouxu** [comment]
One more (general) question... does the linkchecker have any retry mechanism for 5xx errors?

File: book.toml (hunk 2 above)
Comment (camelid) [reply book.toml:47]: Either way, I don't think it really matters because this type of link should always be valid.

**Review by camelid — APPROVED** [review]

**Pull request merged**